### PR TITLE
purge doctest stacktraces

### PR DESCRIPTION
**Pull request by KristofferC**
See discussion in https://github.com/JuliaLang/julia/pull/22075
cc @rfourquet

### CODE REVIEW

**Comment by rfourquet** [comment]
LGTM. What about leaving the `Stacktrace:` line to make explicit that only the stacktrace is omitted?

**Comment by KristofferC** [comment]
I had that first but then I removed it when I saw you had removed it hehe.

**Comment by KristofferC** [comment]
I will add the stacktrace header back.

File: doc/src/manual/documentation.md [hunk 1]
```diff
@@ -128,6 +128,20 @@ As in the example above, we recommend following some simple conventions when wri
    You can then run `make -C doc doctest` to run all the doctests in the Julia Manual, which will
    ensure that your example works.
 
+   To indicate that the output result is truncated, you may write
+   `[...]` at the line where checking should stop. This is useful to
+   hide a stacktrace (which contains non-permanent references to lines
+   of julia code) when the doctest shows that an exception is thrown,
+   for example:
+
+   ````julia
```
Comment (mschauer) [inline doc/src/manual/documentation.md]: is this valid markdown?

Comment (KristofferC) [reply doc/src/manual/documentation.md]: Worked locally.

**Review by StefanKarpinski — APPROVED** [review]
❤️

**Comment by ViralBShah** [comment]
I'm late to this party, but it does appear that the stacktraces are useful.

File: doc/src/manual/documentation.md [hunk 2]
```diff
@@ -128,6 +128,20 @@ As in the example above, we recommend following some simple conventions when wri
...
+   ```jldoctest
+   julia> div(1, 0)
+   ERROR: DivideError: integer division error
+   [...]
+   ```
+   ````
+
    Examples that are untestable should be written within fenced code blocks starting with ````` ```julia`````
```
Comment (mschauer) [inline doc/src/manual/documentation.md]: Ah, got it. So this is wrong, for example.

**Comment by StefanKarpinski** [comment]
Yeah, I don't really get the pushback against this. 99% of the time no one cares about the stack traces so the default should be not including them. In the 1% cases where it's useful, we can still include it.

**Comment by tkelman** [comment]
Only one of these is showing 6 frames deep. The top handful of frames, up to the function that actually gets called, is useful to some people - it tells you where the definition is (especially when nicely linked like in IJulia) that's throwing the error in question. In a perfect world code examples in the docs would be shown as live notebook-style runnable snippets with [most of] the actual output. You might want to collapse long backtraces by default and have them easily expandable with a single click, but static markdown docs don't allow that level of interactivity.

**Comment by ViralBShah** [comment]
`It appears` as in as I looked at it, I felt that things looked better with stacktraces. But, I do not wish to have this PR be blocked because of my slight preference to see the stacktrace. I can imagine that having a 15 line stacktrace is not the best thing in 99% of the cases, but there are certainly cases where the stacktraces are useful to show in the documentation when the error happens.

Maybe, the thing to do is merge this, and introduce them in the 1% cases where they clearly may help.

**Comment by StefanKarpinski** [comment]
Since this is one of *the* main things preventing us from turning doc tests on, we should go ahead with this and if anyone feels that we really need the stack traces in some places, the onus is on them to implement the necessary Documenter.jl feature and insert the desired stack traces.

**Pull request merged**